service/s3/api_op_GetObject.go

```diff
@@ -366,6 +370,180 @@ type GetObjectOutput struct {
 	WebsiteRedirectLocation *string `location:"header" locationName:"x-amz-website-redirect-location" type:"string"`
 }
 
+func (s *GetObjectOutput) UnmarshalAWSPayload(r io.ReadCloser) (err error) {
+	defer func() {
+		if err != nil {
+			*s = GetObjectOutput{}
+		}
+	}()
+	s.Body = r
+	return nil
+}
+
+func (s *GetObjectOutput) UnmarshalAWSREST(r *http.Response) (err error) {
+	defer func() {
+		if err != nil {
+			*s = GetObjectOutput{}
+		}
+	}()
+
+	// First, unmarshal the status code to the status code field in the shape
+
+	// Secondly, unmarshal the http header to the header fields in the shape
+	for k, v := range r.Header {
+		switch k {
+		case http.CanonicalHeaderKey("accept-ranges"):
```

Comment:
I think the `EqualsFold` method may be better because the first method, with `strings.ToLower` requires the metadata prefixes to fall into a `default` case and use a separate switch or if based on prefix. But performance difference between the two might warrant using the strings.ToLower. We'll have to test.